Add bug triage docs (#663)
Green-Software-Foundation/if: CONTRIBUTING.md

> You can contribute anything to the IF, but we are likely to close out unsolicited PRs without merging them. Our issue board is completely open and we have tags (`help-wanted`, `good-first-issue`) to help contributors to choose tasks to work on. We recommend speaking to the core team on Github before starting working on an issue. You can do this by raising an issue or commenting on an existing issue. This helps us to direct your energy is directions that are aligned with our roadmap, prevent multiple people working on the same task, and better manage our board. This all makes it much more likely that your work will get merged.

Reviewer: @jmcook1186 energy is directions --> energy in directions

Author: ```suggestion
You can contribute anything to the IF, but we are likely to close out unsolicited PRs without merging them. Our issue board is completely open and we have tags (`help-wanted`, `good-first-issue`) to help contributors to choose tasks to work on. We recommend speaking to the core team on Github before starting working on an issue. You can do this by raising an issue or commenting on an existing issue. This helps us to direct your energy in directions that are aligned with our roadmap, prevent multiple people working on the same task, and better manage our board. This all makes it much more likely that your work will get merged.
```

## Reporting bugs

We appreciate bug reports! If you experience an issue with IF or one of our plugins, you can report it using our bug reporting template. To do this:

> 1. Go to the [IF repository](https://github.com/Green-Software-Foundation/if) (or [plugin repository](https://github.com/Green-Software-Foundation/if) if you bug relates to a specific plugin)

Reviewer: @jmcook1186 Wrong link for plugin repo
[plugin repository](https://github.com/Green-Software-Foundation/if -->
[plugin repository](https://github.com/Green-Software-Foundation/if-plugins
Maybe the link for unofficial plugins should also be here, or only if-plugins is enough?

Author: ```suggestion
1. Go to the [IF repository](https://github.com/Green-Software-Foundation/if) (or [plugin repository](https://github.com/Green-Software-Foundation/if-plugins) if you bug relates to a specific plugin)
```

2. Click on the `Issues` tab
3. Click on `Create New Issue` and select the `Bug Report` template.
4. Fill out the requested information.

> The more detailed information you provide in then bug report, the easier it will be for us to diagnose, triage and resolve your issue. We ask for some simple information about your issue, including a description of the error, the expected behaviour, the actual behaviour and the stepos we cna take to reproduce the error in our local environments. We also then prompt you to provide a link to [Stackblitz](https://stackblitz.com/) or a similar online environemnt where we can run your mnaifest and observe the error. If you prefer *not* to send a link, we would appreciate a copy of the manifest file that you ran to produce the error, information about your runtime environment and any additional code that's required to reproduce the error. This is all designed to enable us to reproduce the same error and debug it for you as quickly as possible.

Reviewer: @jmcook1186 provide in then bug report --> provide in the bug report

Author: ```suggestion
The more detailed information you provide in the bug report, the easier it will be for us to diagnose, triage and resolve your issue. We ask for some simple information about your issue, including a description of the error, the expected behaviour, the actual behaviour and the stepos we can take to reproduce the error in our local environments. We also then prompt you to provide a link to [Stackblitz](https://stackblitz.com/) or a similar online environment where we can run your manifest and observe the error. If you prefer *not* to send a link, we would appreciate a copy of the manifest file that you ran to produce the error, information about your runtime environment and any additional code that's required to reproduce the error. This is all designed to enable us to reproduce the same error and debug it for you as quickly as possible.
```

| Code not addressing user needs                                                                                         | no product market fit, loss of adoption                                                  | 5        |
| Communication failures within team                                                                                     | Conflicting or duplicating work, frustration, morale damage                              | 4        |
| Communication failures with community                                                                                  | we lose product market fit, we do not have good community retention, reputational damage | 3        |
| Communication failures with leadership                                                                                 | product does not meet business goals                                                     | 3        |
| License compliance failures, including in supply chain (e.g. exposing privileged api responses for free via  a plugin) | 4                                                                                        |

> | Bugs affecting releases                                                                                                | users stuck on old versions                                                              | 2        |

Reviewer: @jmcook1186 in IF-docs I wrote a question about the severity level of "bugs affecting releases" . In case it is changed, please update it here as well

Author: ```suggestion
| Bugs affecting releases                                                                                                | users stuck on old versions                                                              | 4        |
```

| Strategy failures                                                                                                      | no product market fit                                                                    | 2        |

The mapping of severity to label is as follows:

| Severity | Label |
| -------- | ----- |
| 1        | L     |
| 2        | M     |
| 3        | M     |
| 4        | H     |
| 5        | H     |

During the bug triage we will also discuss a remediation plan for the bug. This will be communicated in the comments on the bug report. For high urgency bugs, the fix will be implemented as soon as possible, maybe reorganizing our current work to accommodate it. For medium priority bugs, we will schedule the fix in the next available sprint. Low priority bugs will be backlogged and addressed when there is developer time available. Low priority bugs will also be tagged `help-wanted` so that they can be addressed by community members.

Not every bug will be fixed. We may decide *not* to fix a bug in cases such as:

- fixing the bug has some detrimental side effect elsewhere in the product
- the bug has a fix coming soon as part of another upgrade

> - the bug is only problematic for a single specific use casea nd fixing it would break features relied upon by other users

Reviewer: @jmcook1186 casea nd fixing --> case and fixing

Author: ```suggestion
- the bug is only problematic for a single specific use case and fixing it would break features relied upon by other users
```
